src/ca65/main.c

```diff
@@ -1040,7 +1048,14 @@ int main (int argc, char* argv [])
     /* If no CPU given, use the default CPU for the target */
     if (GetCPU () == CPU_UNKNOWN) {
         if (Target != TGT_UNKNOWN) {
-            SetCPU (GetTargetProperties (Target)->DefaultCPU);
+            int cpu;
+
+            cpu = GetTargetProperties (Target)->DefaultCPU;
+
+            if (cpu == CPU_UNKNOWN) {
+              Error ("CPU of target is unknown");
+            }
+            SetCPU (cpu);
```

Comment:
I had an error in this place at some time and had to check the code what 
exactly went wrong.
Some existing error messages give no idea what the problem could be..

On 09.05.2015 21:05, greg-king5 wrote:
>
> In src/ca65/main.c 
> <https://github.com/cc65/cc65/pull/157#discussion_r29995407>:
>
> > @@ -1040,7 +1048,14 @@ int main (int argc, char* argv [])
> >      /* If no CPU given, use the default CPU for the target */
> >      if (GetCPU () == CPU_UNKNOWN) {
> >          if (Target != TGT_UNKNOWN) {
> > -            SetCPU (GetTargetProperties (Target)->DefaultCPU);
> > +            int cpu;
> > +
> > +            cpu = GetTargetProperties (Target)->DefaultCPU;
> > +
> > +            if (cpu == CPU_UNKNOWN) {
> > +              Error ("CPU of target is unknown");
> > +            }
> > +            SetCPU (cpu);
>
> That test isn't needed. None of the |DefaultCPU| entries is allowed to 
> have |CPU_UNKNOWN| -- even the |none| target has a valid CPU.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/cc65/cc65/pull/157/files#r29995407>.
>

-- 
Mit freundlichen Grüssen

Rainer Keuchel